tests/test_fetch.py

```diff
@@ -63,3 +63,24 @@ def test_fetch_with_scheme_not_present():
         url = "abc://speedtest/1KB.zip"
         response = fetch(url=url)
         assert "Not a supported/known scheme." == e_info
+
+
+@mock.patch("fetchcode.resolve_url_from_purl")
```

Comment:
The code is not properly tested if mocking `resolve_url_from_purl`
Instead, mock `_http_exists` and use a supported PURL.
Also make sure to test with a purl that cannot be resolved.